packages/components/src/internal/components/search/SampleFinderSection.tsx

```diff
@@ -138,21 +189,73 @@ export const SampleFinderSection: FC<Props> = memo(props => {
 });
 
 interface SampleFinderSamplesProps extends SampleFinderSamplesGridProps {
-    cards: FilterCardProps[];
+    cards: FilterProps[];
+    filterChangeCounter: number;
+    selectionKeyPrefix: string;
 }
 
 export const SampleFinderSamplesImpl: FC<SampleFinderSamplesGridProps & InjectedQueryModels> = memo(props => {
-    const { actions, queryModels } = props;
+    const { actions, queryModels, gridButtons, excludedCreateMenuKeys } = props;
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+
+    useEffect(() => {
+        const allModelsLoaded = Object.values(queryModels).filter(model => model.isLoading).length == 0;
+        if (allModelsLoaded && isLoading) {
+            const promises = [];
+            Object.values(queryModels).forEach(queryModel => {
+                const {hasUpdates, columns} = getFinderViewColumnsConfig(queryModel);
+                if (hasUpdates) {
+                    promises.push(saveFinderGridView(queryModel.schemaQuery, columns));
+                }
+            });
+            Promise.all(promises).then((schemaQueries) => {
+                // since we have updated views, we need to invalidate the details cache so we pick up the new views
+                schemaQueries.forEach(schemaQuery => {
+                    invalidateQueryDetailsCache(schemaQuery);
+                });
+                setIsLoading(false);
+            }).catch(reason => {
+                console.error("Error saving all finder views.", reason);
+                setIsLoading(false);
+            });
+        }
+    }, [queryModels]);
+
+
+    useEffect(() => {
+        return () => {
+            if (queryModels) {
+                for (const queryModel of Object.values(queryModels)) {
+                    (async () => {
+                        try {
+                            await removeFinderGridView(queryModel);
```

Comment:
nice, these calls to `saveFinderGridView` and `removeFinderGridView` ended up being pretty clean in the end.